spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -633,9 +633,15 @@ private void considerAddingAConsumer() {
 		}
 	}
 
+	private int getConsumersThatHaveNotInitiatedStopping() {
+		synchronized (this.consumersMonitor) {
+			return (int) this.consumers.values().stream().filter(x -> x = Boolean.TRUE).count();
```

Comment:
Hi @artembilan !

First of all, sorry for the lousy filter. I have put in the requested changes and actually I agree with them. I think it looks decent now. After these changes however, the question arises - Why do we even need a key value pair here?  

I mean just a Set<BlockingQueueConsumer> activeConsumers would be good enough to keep track of the number of active consumers as well as whether a particular consumer is still active (if its a member of the set). That way we also simplify the logic of `isActive(BlockingQueueConsumer)` as it will be pretty much an `activeConsumers.contains()` call.